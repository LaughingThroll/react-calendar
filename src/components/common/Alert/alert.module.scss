@import 'styles/utils/variables';

.alertWrapper {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
}

.alert {
  position: absolute;
  top: 25%;
  left: 50%;
  transform: translateX(-50%);
  max-width: 350px;
  width: 100%;
  background-color: $alert-bg;
  border-radius: 4px;
  padding: 20px 30px;
  border: 1px solid $alert-border;
  z-index: 10;
  &__message {
    max-width: 300px;
    text-align: center;
    margin-bottom: 30px;
    font-size: 1.5em;
    line-height: 1.3em;
    &_error {
      color: $alert-message-error;
    }
    &_success {
      color: $alert-message-success;
    }
    &_warning {
      color: $alert-message-warning;
    }
  }
  &__button {
    margin: 0 auto;

    &_error {
      background-color: $alert-message-error;
      &:hover {
        background-color: darken($alert-message-error, 10);
      }
    }
    &_success {
      background-color: $alert-message-success;
      &:hover {
        background-color: darken($alert-message-success, 10);
      }
    }
    &_warning {
      background-color: $alert-message-warning;
      &:hover {
        background-color: darken($alert-message-warning, 10);
      }
    }
  }
}
