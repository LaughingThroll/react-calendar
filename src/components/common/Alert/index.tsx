import React from 'react'
import classnames from 'classnames'
import styles from './alert.module.scss'

import { Button } from '../'
import { TChildren } from '../../../types/utilsTypes'

export interface AlertProps {
  onAccept: () => void
  message: TChildren
  success?: boolean
  error?: boolean
  warning?: boolean
  classNames?: string[]
  buttonText?: TChildren
}

const {
  alert,
  alertWrapper,
  alert__message,
  alert__message_error,
  alert__message_success,
  alert__message_warning,
  alert__button,
  alert__button_error,
  alert__button_success,
  alert__button_warning,
} = styles

const Alert: React.FC<AlertProps> = ({
  message = 'Alert fake',
  onAccept,
  buttonText = 'Okey',
  error,
  success,
  warning,
  classNames = [],
}) => {
  return (
    <div className={alertWrapper}>
      <div className={[alert, ...classNames].join(' ')}>
        <p
          className={classnames({
            [alert__message]: true,
            [alert__message_error]: error,
            [alert__message_success]: success,
            [alert__message_warning]: warning,
          })}
        >
          {message}
        </p>
        <Button
          onClick={onAccept}
          classNames={classnames({
            [alert__button]: true,
            [alert__button_error]: error,
            [alert__button_success]: success,
            [alert__button_warning]: warning,
          }).split(' ')}
        >
          {buttonText}
        </Button>
      </div>
    </div>
  )
}

export default React.memo(Alert)
