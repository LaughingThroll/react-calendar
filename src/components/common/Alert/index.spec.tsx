import { shallow, render } from 'enzyme'
import Alert, { AlertProps } from './'

const setUp = (props: AlertProps) => shallow(<Alert {...props} />)

describe('Alert render', () => {
  it('default render', () => {
    const props: AlertProps = { onAccept: jest.fn(), message: 'alert' }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('alert')).toBe(true)
  })

  it('default render with message compnent', () => {
    const MessageComponent = () => {
      return <div>Some Text</div>
    }

    const props: AlertProps = { onAccept: jest.fn(), message: <MessageComponent /> }
    const tree = render(<Alert {...props} />)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('Some Text')).toBe(true)
  })

  it('render with props error', () => {
    const props: AlertProps = { onAccept: jest.fn(), message: 'error', error: true }
    const tree = render(<Alert {...props} />)

    expect(tree.text().includes('error')).toBe(true)
    expect(tree.find('.alert__message').hasClass('alert__message_error')).toBe(true)
    expect(tree.find('button').hasClass('alert__button_error')).toBe(true)
  })

  it('render with props success', () => {
    const props: AlertProps = { onAccept: jest.fn(), message: 'success', success: true }
    const tree = render(<Alert {...props} />)

    expect(tree.text().includes('success')).toBe(true)
    expect(tree.find('.alert__message').hasClass('alert__message_success')).toBe(true)
    expect(tree.find('button').hasClass('alert__button_success')).toBe(true)
  })

  it('render with props warning', () => {
    const props: AlertProps = { onAccept: jest.fn(), message: 'warning', warning: true }
    const tree = render(<Alert {...props} />)

    expect(tree.text().includes('warning')).toBe(true)
    expect(tree.find('.alert__message').hasClass('alert__message_warning')).toBe(true)
    expect(tree.find('button').hasClass('alert__button_warning')).toBe(true)
  })
})

describe('Alert onAccept', () => {
  it('onAccept must be call by click', () => {
    const props: AlertProps = { onAccept: jest.fn(), message: 'warning', warning: true }
    const tree = render(<Alert {...props} />)

    expect(tree.text().includes('warning')).toBe(true)
    expect(tree.find('.alert__message').hasClass('alert__message_warning')).toBe(true)
    expect(tree.find('button').hasClass('alert__button_warning')).toBe(true)
  })
})
