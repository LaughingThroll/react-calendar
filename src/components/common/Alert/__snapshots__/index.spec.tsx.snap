// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert render default render 1`] = `
<div
  className="alertWrapper"
>
  <div
    className="alert"
  >
    <p
      className="alert__message"
    >
      alert
    </p>
    <Memo(Button)
      classNames={
        Array [
          "alert__button",
        ]
      }
      onClick={[MockFunction]}
    >
      Okey
    </Memo(Button)>
  </div>
</div>
`;

exports[`Alert render default render with message compnent 1`] = `
<div
  class="alertWrapper"
>
  <div
    class="alert"
  >
    <p
      class="alert__message"
    />
    <div>
      Some Text
    </div>
    <p />
    <button
      class="button button_primary alert__button"
      type="button"
    >
      Okey
    </button>
  </div>
</div>
`;
